File: scripts/test_solutions/347_top_k_frequent.js
var topKFrequent = function(nums, k) {
    const freq = new Map();
    for (const n of nums) freq.set(n, (freq.get(n) || 0) + 1);
    const bucket = Array.from({length: nums.length + 1}, () => []);
    for (const [num, count] of freq) bucket[count].push(num);
    const result = [];
    for (let i = bucket.length - 1; i >= 0 && result.length < k; i--) {
        result.push(...bucket[i]);
    }
    return result.slice(0, k);
};

File: scripts/test_solutions/347_top_k_frequent.py
def topKFrequent(nums, k):
    freq = {}
    for n in nums:
        freq[n] = freq.get(n, 0) + 1
    bucket = [[] for _ in range(len(nums) + 1)]
    for num, count in freq.items():
        bucket[count].append(num)
    result = []
    for i in range(len(bucket) - 1, -1, -1):
        result.extend(bucket[i])
        if len(result) >= k:
            break
    return result[:k]
